meta-lmp-base/recipes-containers/docker/files/0001-tarexport-Load-image-without-layer-copying.patch

```diff
@@ -0,0 +1,106 @@
+From 793434dc979691b7f64b75367d44e38deef54e3f Mon Sep 17 00:00:00 2001
+From: Mike Sul <mike.sul@foundries.io>
+Date: Fri, 22 Sep 2023 12:49:07 +0200
+Subject: [PATCH] tarexport: Load image without layer copying
+
+Signed-off-by: Mike <mike.sul@foundries.io>
+---
+ image/tarexport/load.go      | 39 ++++++++++++++++++++++++++++--------
+ image/tarexport/tarexport.go |  1 +
+ 2 files changed, 32 insertions(+), 8 deletions(-)
+
+diff --git a/image/tarexport/load.go b/image/tarexport/load.go
+index d9f3873081..6fb2f76983 100644
+--- a/image/tarexport/load.go
++++ b/image/tarexport/load.go
+@@ -70,7 +70,7 @@ func (l *tarexporter) Load(inTar io.ReadCloser, outStream io.Writer, quiet bool)
+ 	var imageRefCount int
+ 
+ 	for _, m := range manifest {
+-		configPath, err := safePath(tmpDir, m.Config)
++		configPath, err := safePathWithTmp(m.LayerRootDir, tmpDir, m.Config)
+ 		if err != nil {
+ 			return err
+ 		}
+@@ -93,7 +93,7 @@ func (l *tarexporter) Load(inTar io.ReadCloser, outStream io.Writer, quiet bool)
+ 		}
+ 
+ 		for i, diffID := range img.RootFS.DiffIDs {
+-			layerPath, err := safePath(tmpDir, m.Layers[i])
++			layerPath, err := safePathWithTmp(m.LayerRootDir, tmpDir, m.Layers[i])
+ 			if err != nil {
+ 				return err
+ 			}
+@@ -123,14 +123,21 @@ func (l *tarexporter) Load(inTar io.ReadCloser, outStream io.Writer, quiet bool)
+ 		for _, repoTag := range m.RepoTags {
+ 			named, err := reference.ParseNormalizedNamed(repoTag)
+ 			if err != nil {
+-				return err
++				return fmt.Errorf("Invalid image reference: %s", repoTag)
+ 			}
+-			ref, ok := named.(reference.NamedTagged)
+-			if !ok {
+-				return fmt.Errorf("invalid tag %q", repoTag)
++			switch ref := named.(type) {
++				case reference.NamedTagged: {
++					l.setLoadedTag(ref, imgID.Digest(), outStream)
++					outStream.Write([]byte(fmt.Sprintf("Loaded image with tag: %s\n", reference.FamiliarString(ref))))
++				}
++				case reference.Canonical: {
```

Comment:
@doanac Just heads up. This allows us to pass a digest/hashed URI and set for the loaded/exported image in the repository file.